docs/vector-indexes.md

```diff
@@ -62,7 +62,7 @@ class MyPage(VectorIndexedMixin, Page):
 
 ### Indexing across models
 
-One of the things you might want to do with a custom index is query across multiple models, or on a subset of models. To do this, they need to be in a vector index together.
+One of the things you might want to do with a custom index is query across multiple models, or on a subset of models. To do this, models must share a common concrete parent model (e.g. Wagtail's `Page` model), then they can be included together in a custom vector index.
```

Comment:
I thought mentioning the 'shared concrete parent model' thing was important here, as I expect that to be the key limiting factor going forward (you need to override a lot more than `get_documents()` otherwise).